src/Util/YamlSourceManipulator.php

```diff
@@ -449,8 +457,15 @@ private function changeValueInYaml($value)
 
         $endValuePosition = $this->findEndPositionOfValue($originalVal);
 
-        $newYamlValue = $this->convertToYaml($value);
-        if (!\is_array($originalVal) && \is_array($value)) {
+        $isMultilineValue = null !== $this->findPositionOfMultilineCharInLine($this->currentPosition);
+        $originalValEndsWithNewLine = $isMultilineValue ? rtrim($originalVal, "\n") !== $originalVal : false;
```

Comment:
I think the `rtrim($originalVal, "\n") !== $originalVal` is confusing me. What about this instead?

`str_ends_with($originalValue, "\n")`

That's a PHP 8 function, but if you require `"symfony/polyfill-php80": "^1.15"`, then we can use it.

Reply:
Are you OK to require `symfony/polyfill-php80` then ? I'll made apprioriate changes if so (there are some other lines like https://github.com/symfony/maker-bundle/pull/610/files#diff-a07d297f64d0bf4a1e9c902b787ecf20R219-R223)
```
if (
    $this->isMultilineString($newVal)
    && rtrim($currentVal, "\n") !== $currentVal
    && rtrim($newVal, "\n") === $newVal
)
```